app/src/main/java/com/codesoom/assignment/controllers/TaskController.java

```diff
@@ -0,0 +1,86 @@
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.models.Task;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/tasks")
+@CrossOrigin
+public class TaskController {
+    private List<Task> tasks = new ArrayList<>();
+    private Long newId = 0L;
+
+    @GetMapping
+    public List<Task> listTask() {
+        return tasks;
+    }
+
+    @GetMapping("/{id}")
+    public Task findTask(@PathVariable Long id) {
+        Optional<Task> task = Optional.ofNullable(findbyId(id));
+
+        if (task.isEmpty()) {
+            return null;
+        }
+
+        return task.get();
+    }
+
+    @RequestMapping(value = "/{id}", method = {RequestMethod.PUT, RequestMethod.POST})
+    public Task updateTask(@RequestBody Task everything, @PathVariable Long id) {
+        Task task = findbyId(id);
+        //if절로 NullPointerException 예외 처리
+        if (task == null) {
+            System.out.println("java.lang.NullPointerException");
+        }
+
+        if (task != null) {
+            task.setTitle(everything.getTitle());
+        }
+
+        return task;
+    }
+
+    @PostMapping
+    public Task createTask(@RequestBody Task task) {
+        task.setId(generateId());
+        tasks.add(task);
+
+        return task;
+    }
+
+    @DeleteMapping("/{id}")
+    public Task deleteTask(@PathVariable Long id) {
+        Task task = findbyId(id);
+        tasks.remove(task);
+
+        return task;
+    }
+
+    private Long generateId() {
+        newId += 1;
+
+        return newId;
+    }
+
+    private Task findbyId(Long id) {
```

Comment:
`Optional`을 무의미하게 사용하셨어요. 지금 작성하신 코드에서는 `Optional`을 썼지만 메소드가 `null`을 리턴할 수 있으므로 `Optional`을 안 쓴 것과 다름이 없습니다. 이 상황에서는 메소드 시그니처가 이렇게 되도록 작업해 보세요.
```suggestion
    private Optional<Task> findbyId(Long id) {
```

Reply:
Optional에 대해서 질문 한가지 드려 될까요?

`Optional<Task> task`을 사용 했을때 와 `Task task` 사용했을때 models/Task 쪽에서 호출이 달라지다 보니 `Optional<Task> task`사용 했을 경우 Task 객체를 Optional 적용해서 따로 객체를 생성해야 할까요??